### ISSUE TITLE
measureText() doesn't work when textAlign="center"

### ISSUE THREAD

**Issue by dan-mba** [samizdatco/skia-canvas#36]
I am trying to migrate from node-canvas and am having problems with measureText(). 

Here is a simple testcase showing the problem. 
```
const canvas = new Canvas(2048, 2048);
const ctx = canvas.getContext("2d");
console.log(ctx.measureText("hello"));
ctx.textAlign = "center";
console.log(ctx.measureText("hello"));
```

textAlign default width = 21.59
textAlign center width  = 0

I am not sure if the problem is this package or one of its dependencies

here is the full output:
```
TextMetrics {
  width: 21.59000015258789,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 21.59000015258789,
  actualBoundingBoxAscent: 10.791015625,
  actualBoundingBoxDescent: 2.509765625,
  fontBoundingBoxAscent: 10.791015625,
  fontBoundingBoxDescent: 2.509765625,
  emHeightAscent: 10.791015625,
  emHeightDescent: 2.509765625,
  hangingBaseline: 7.001953125,
  alphabeticBaseline: 0,
  ideographicBaseline: -2.509765625,
  lines: [
    {
      x: 0,
      y: -10.791015625,
      width: 21.59000015258789,
      height: 13.30078125,
      baseline: 0,
      startIndex: 0,
      endIndex: 6
    }
  ]
}
TextMetrics {
  width: 0,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 0,
  actualBoundingBoxAscent: -0,
  actualBoundingBoxDescent: 0,
  fontBoundingBoxAscent: 10.791015625,
  fontBoundingBoxDescent: 2.509765625,
  emHeightAscent: 10.791015625,
  emHeightDescent: 2.509765625,
  hangingBaseline: 7.001953125,
  alphabeticBaseline: 0,
  ideographicBaseline: -2.509765625,
  lines: [
    {
      x: 99967.609375,
      y: -10.791015625,
      width: -99956.8125,
      height: 13.30078125,
      baseline: 0,
      startIndex: 0,
      endIndex: 6
    }
  ]
}
```

**Comment by dan-mba** [comment]
To gather comparison data I ran a test using `node-canvas`.
I changed the font to "12px Arial" to get rid of a warning from `node-canvas`.

Running the test on `node-canvas` , the results were:
```
{
  width: 27,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 27,
  actualBoundingBoxAscent: 9,
  actualBoundingBoxDescent: 0,
  emHeightAscent: 12,
  emHeightDescent: 3,
  alphabeticBaseline: -0
}
{
  width: 27,
  actualBoundingBoxLeft: 13.5,
  actualBoundingBoxRight: 40.5,
  actualBoundingBoxAscent: 9,
  actualBoundingBoxDescent: 0,
  emHeightAscent: 12,
  emHeightDescent: 3,
  alphabeticBaseline: -0
}
```


Re-running the test  on `skia-canvas` using Arial, the results were:
```
TextMetrics {
  width: 25.350000381469727,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 25.350000381469727,
  actualBoundingBoxAscent: 10.86328125,
  actualBoundingBoxDescent: 2.54296875,
  fontBoundingBoxAscent: 11.682819366455078,
  fontBoundingBoxDescent: 2.7171804904937744,
  emHeightAscent: 11.682819366455078,
  emHeightDescent: 2.7171804904937744,
  hangingBaseline: 8.40234375,
  alphabeticBaseline: 0,
  ideographicBaseline: -2.7171804904937744,
  lines: [
    {
      x: 0,
      y: -10.86328125,
      width: 25.350000381469727,
      height: 13.40625,
      baseline: 0,
      startIndex: 0,
      endIndex: 6
    }
  ]
}
TextMetrics {
  width: 0,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 0,
  actualBoundingBoxAscent: -0,
  actualBoundingBoxDescent: 0,
  fontBoundingBoxAscent: 11.682819366455078,
  fontBoundingBoxDescent: 2.7171804904937744,
  emHeightAscent: 11.682819366455078,
  emHeightDescent: 2.7171804904937744,
  hangingBaseline: 8.40234375,
  alphabeticBaseline: 0,
  ideographicBaseline: -2.7171804904937744,
  lines: [
    {
      x: 99961.96875,
      y: -10.86328125,
      width: -99949.296875,
      height: 13.40625,
      baseline: 0,
      startIndex: 0,
      endIndex: 6
    }
  ]
}
```

**Comment by samizdatco** [comment]
Thanks for doing all that testing. I believe it's working as intended now that I've corrected the rectangle arithmetic in d9fa464.

Here's some sample code:
```js
const canvas = new Canvas(400, 200),
      ctx = canvas.getContext("2d"),
      msg = "hamburgefonstiv";

let x = 200,
    y = 50;

ctx.strokeStyle = 'skyblue'
ctx.beginPath()
ctx.moveTo(x, 0)
ctx.lineTo(x, canvas.height)
ctx.stroke()

for (const align of ['left', 'center', 'right']){
  ctx.font = '22px Arial'
  ctx.textAlign = align
  ctx.fillText(msg, x, y)

  let metrics = ctx.measureText(msg),
      line = metrics.lines[0];
  ctx.strokeStyle = 'red'
  ctx.beginPath()
  ctx.rect(x + line.x, y + line.y, line.width, line.height)
  ctx.stroke()

  console.log(align, metrics.width)

  y += 50
}

canvas.saveAs("alignments.png", {density:2})
```

Which prints to the console:
```console
left 162.64999389648438
center 162.6484375
right 162.6484375
```

And uses the measurements to draw these red frames around the text:
<img width="400" alt="out" src="https://user-images.githubusercontent.com/469523/130331763-e084b2e9-053c-4103-878c-0bc5ad61f810.png">

**Comment by dan-mba** [comment]
Thanks for the quick turnaround.

When do you expect to have a release that includes the fix?
